### ISSUE TITLE
Resonance requires a Music folder re-add after a reboot

### ISSUE THREAD

**Issue by tfuxu** [nate-xyz/resonance#47]
From issue #24,

Resonance doesn't handle persistent access to allowed folders (if a folder is located on a separate drive) on Flatpak builds. Currently it requires from the user to remove the folder in preferences window and to add it again in order to properly playback tracks.

**Comment by nate-xyz** [comment]
Is the drive mounted when you play the track? Because my music collection is also on a separate drive and I have not come across this issue

**Comment by tfuxu** [comment]
Yes, it is mounted during a playback. I don't know if this could be important, but that drive is using btrfs and the system one is using ext4.

**Comment by nate-xyz** [comment]
Is the path uri the same every time? Idk how btrfs handles mounted drive paths

**Comment by tfuxu** [comment]
> Is the path uri the same every time? Idk how btrfs handles mounted drive paths

I'll need to check that

**Comment by honnip** [comment]
> Is the path uri the same every time?

no, a different random hash is used each time

```shell
> ls /run/user/1000/doc/73a42925/
> ls /run/user/1000/doc/9e5ffb01/
> ls /run/user/1000/doc/f9bc6ed6/
Music/
```

![스크린샷 2023-04-10 09-10-46](https://user-images.githubusercontent.com/108175486/230817254-391ef7e1-8f3e-41d7-b3a6-9d346c5098c7.png)

**Comment by tfuxu** [comment]
It changed for me too after a reboot:
1st reboot:
```
/run/user/1000/doc/42c77e98/Music
```

2nd reboot:
```
/run/user/1000/doc/b606c4ff/Music
```

**Comment by nate-xyz** [comment]
To address this issue I probably need to identify mounted drives by UUID and not rely on uris. I wonder how other applications deal with btrfs, this must be a common issue?

**Comment by honnip** [comment]
I think this is the behavior of portal. See [portal.FileChooser](https://flatpak.github.io/xdg-desktop-portal/#gdbus-org.freedesktop.portal.FileChooser), [$XDG_RUNTIME_DIR](https://www.freedesktop.org/software/systemd/man/pam_systemd.html#%24XDG_RUNTIME_DIR)

This can be solved by using [FileChooserDialog](https://gtk-rs.org/gtk4-rs/stable/latest/docs/gtk4/struct.FileChooserDialog.html) than [FileChooserNative](https://gtk-rs.org/gtk4-rs/stable/latest/docs/gtk4/struct.FileChooserNative.html) to avoid portal. (requires filesystem=host (and device=all maybe?) to flatpak package)

**Comment by honnip** [comment]
As I understand these docs, the reason I said to avoid portal is that files opened through portal are proxied to `/run/user/$UID/doc`. ([portal.FileChooser](https://flatpak.github.io/xdg-desktop-portal/#gdbus-org.freedesktop.portal.FileChooser))

> The selected files will be made accessible to the application via the document portal, and the returned URI will point into the document portal fuse filesystem in /run/user/$UID/doc/. 

Because `/run/user/$UID` is `$XDG_RUNTIME_DIR`, its contents are lost on every boot. ([file-hierarchy#/run/user](https://www.freedesktop.org/software/systemd/man/file-hierarchy.html#/run/user/), [pam_systemd#$XDG_RUNTIME_DIR](https://www.freedesktop.org/software/systemd/man/pam_systemd.html#%24XDG_RUNTIME_DIR))

>It is automatically created the first time a user logs in and removed on the user's final logout.



so it's not a problem with native

**Comment by gitjod** [comment]
Altering the Filesystem permissions via Flatseal fixed this issue for me.

Here's what I currently have:

![image](https://github.com/nate-xyz/resonance/assets/12417441/8dd86e2f-b3f9-4f0b-80c5-2fb350dc8da1)
